tools/kraken.wdl

```diff
@@ -0,0 +1,336 @@
+        memory_gb: "RAM to allocate for task"
+        added_disk_size_gb: "Additional disk space to allocate for task. Default disk size is determined dynamically based on `fastas` size."
+        max_retries: "Number of times to retry in case of failure"
+    }
+
+    Int fastas_size = ceil(size(fastas, "GiB"))
+    Int disk_size_gb = fastas_size * 5 + added_disk_size_gb
+
+    command <<<
+        set -euo pipefail
+
+        >&2 echo "*** start adding custom FASTAs ***"
+        echo "~{sep="\n" fastas}" > fastas.txt
+        while read -r fasta; do
+            gunzip -c "$fasta" > tmp.fa
+            kraken2-build --add-to-library tmp.fa --db ~{db_name}
+        done < fastas.txt
+        rm tmp.fa
+        >&2 echo "*** done adding custom FASTAs ***"
+
+        tar -C ~{db_name}/ -czf "~{db_name}.tar.gz" .
+
+        rm -r ~{db_name}
+    >>>
+ 
+    runtime {
+        memory: memory_gb + " GB"
+        disk: disk_size_gb + " GB"
+        cpu: 1
+        docker: 'quay.io/biocontainers/kraken2:2.1.2--pl5321h9f5acd7_2'
+        maxRetries: max_retries
+    }
+
+    output {
+        File custom_library = db_name + ".tar.gz"
+    }
+
+    meta {
+        author: "Andrew Frantz"
+        email: "andrew.frantz@stjude.org"
+        description: "This WDL task adds custom FASTAs to a Kraken2 DB."
+    }
+}
+
+task build_db {
+    input {
+        Array[File] tarballs
+        String db_name = "kraken2_db"
+        Int kmer_len = 35
+        Int minimizer_len = 31
+        Int minimizer_spaces = 7
+        Int max_db_size_gb = -1
+        Float load_factor = 0.7
+        Int added_memory_gb = 0
+        Int added_disk_size_gb = 0
+        Int ncpu = 1
+        Boolean detect_nproc = false
+        Int max_retries = 1
+    }
+
+    parameter_meta {
+        tarballs: "Tarballs containing the kraken taxonomy and at least one library. Tarballs must not have a root directory."
+        db_name: "Name for output tarball. The suffix `.tar.gz` will be added."
+        kmer_len: "K-mer length in bp"
+        minimizer_len: "Minimizer length in bp"
+        minimizer_spaces: "Number of characters in minimizer that are ignored in comparisons"
+        max_db_size_gb: "Maximum number of GBs for Kraken 2 hash table; if the estimator determines more would normally be needed, the reference library will be downsampled to fit."
+        load_factor: "Proportion of the hash table to be populated (must be between 0 and 1)"
+        added_memory_gb: "Additional RAM to allocate for task. Default RAM is allocated dynamically based on the database size."
+        added_disk_size_gb: "Additional disk space to allocate for task. Default disk size is determined dynamically based on `tarballs` size."
+        ncpu: "Number of cores to allocate for task"
+        detect_nproc: "Use all available cores"
```

Comment:
We may want to globally update this with a second sentence like "This is useful in systems where a node meeting minimum requirements is allocated (e.g. cloud providers), rather than a fixed number of cores (e.g. cluster environments". That feels very wordy, especially to repeat everywhere, but makes the intention clear.

Reply:
Maybe just a short option like "Recommended for cloud environments"

Reply:
```suggestion
        detect_nproc: "Use all available cores. Recommended for cloud environments. Not recommended for cluster environments."
```
?